### DISCUSSION TITLE
Composer bisect command

### DISCUSSION THREAD

**Discussion by mad-briller** [composer/composer#11072] [General]
This is just a random idea i've had while upgrading projects and having a hard time narrowing down issues, not sure if issues or discussions is the right place to put this.

Even with all the will in the world to respect semver, packages inevitability break backwards compatibility in ways they did not forsee, everyone is human afterall.

Chasing the version that introduced the issue can be quite hard currently, as you have to load up packagist and look at released versions and install them one by one to see if they are the issue.
This reduces the likelihood that a developer will report the issue to the package maintainers, and also makes the developer less likely to actually update the package, as its safer to stay on the "last working version" and put an explicit version number in `composer.json`.

When a incorrect change is introduced in git, `git bisect` makes it super easy to chase the commit that introduced the issue. It would be great if composer had a `composer bisect <package>` that worked similarly. 
This would make narrowing down which version introduced a bc break much easier. 

Thanks for your time.

**Comment by Seldaek** [comment]
Created https://github.com/composer/composer/issues/11119 to keep track of it, IMO it's a good idea.
